### PR TITLE
Update tx view to render inline datums and reference inputs

### PR DESCRIPTION
**Pull request by Jimbo4350**
Resolves: #4045 and #4039

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Run/Friendly.hs [hunk 1]
```diff
@@ -452,6 +456,10 @@ friendlyAuxScripts = \case
   TxAuxScriptsNone -> Null
   TxAuxScripts _ scripts -> String $ textShow scripts
 
+friendlyReferenceInputs :: TxInsReference build era -> Aeson.Value
```
Comment (JaredCorduan) [inline cardano-cli/src/Cardano/CLI/Run/Friendly.hs:459]: json = friendly?

**Review by JaredCorduan — APPROVED** [review]
LGTM!

**Review by erikd — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Run/Friendly.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Run/Friendly.hs:459]: Apparently. I was just following the pattern implemented by Yuri. All of it needs to be refactored though.

**Pull request merged**